pylabrobot/pumps/pump.py

```diff
@@ -1,27 +1,71 @@
-from pylabrobot.machine import MachineFrontend
-
 from .backend import PumpBackend
+from .calibration import PumpCalibration
+from typing import Optional, Union
+from pylabrobot.machine import MachineFrontend
+import asyncio
 
 
 class Pump(MachineFrontend):
-  """ Frontend for a (peristaltic) pump. """
+  """ Frontend for a (peristaltic) pump.
+
+  Attributes:
+      backend: The backend that the pump is controlled through.
+      calibration: The calibration of the pump.
+
+      """
 
-  def __init__(self, backend: PumpBackend):
+  def __init__(self, backend: PumpBackend, calibration: Optional[PumpCalibration] = None):
     self.backend: PumpBackend = backend
+    self.calibration = calibration
```

Comment:
Perhaps it's worth getting the single calibration from the object know, so we don't have to index it below for each use. It may also be good to raise an error if more than one item exists in the calibration list.

Reply:
adjusted this to call the first item